packages/core/src/anchor/anchor-processing-loop.ts

```diff
@@ -50,6 +58,10 @@ export class AnchorProcessingLoop {
       concurrency,
       store.infiniteList(batchSize),
       async (streamId) => {
+        // Exit early if we've already polled this stream in the last pacing interval
+        if (!this.checkPollTime(streamId.toString())) {
```

Comment:
we shouldn't cache by streamid, we should do it by CID of the commit we are polling for.  Otherwise if the stream gets updated we risk delaying the creation of the new request against the CAS.

Reply:
Is that a big problem? We also do a fire-and-forget submission to the CAS before adding a CID to the processing loop. I don't have a strong opinion either way, just feels less noisy to create a request / get status per stream versus per CID.

Reply:
feels more error prone to me.  You also have to make sure to remove entries from the cache if you're caching by streamid, whereas you can just let them expire naturally if you're only caching by CID.  Also CIDs are smaller than StreamIDs, so will keep the cache size down.

But no, it's not a big deal either way, if you prefer streamid I don't feel that strongly.